source/common/http/utility.cc

```diff
@@ -185,12 +185,10 @@ bool Utility::isInternalRequest(const HeaderMap& headers) {
 
 bool Utility::isWebSocketUpgradeRequest(const HeaderMap& headers) {
   return (headers.Connection() && headers.Upgrade() &&
-          (0 == StringUtil::caseInsensitiveCompare(
-                    headers.Connection()->value().c_str(),
-                    Http::Headers::get().ConnectionValues.Upgrade.c_str())) &&
-          (0 == StringUtil::caseInsensitiveCompare(
-                    headers.Upgrade()->value().c_str(),
-                    Http::Headers::get().UpgradeValues.WebSocket.c_str())));
+          headers.Connection()->value().caseInsensitiveContains(
+                  Http::Headers::get().ConnectionValues.Upgrade.c_str()) &&
+          headers.Upgrade()->value().caseInsensitiveContains(
+                  Http::Headers::get().UpgradeValues.WebSocket.c_str()));
```

Comment:
Nit: please copy paste the explanation you had for doing the token comparison (about Firefox sending multiple tokens in the connection string)